app/src/main/java/fr/neamar/kiss/cache/MemoryCacheHelper.java

```diff
@@ -0,0 +1,165 @@
+package fr.neamar.kiss.cache;
+
+import android.content.ComponentName;
+import android.content.Context;
+import android.graphics.drawable.Drawable;
+import android.os.AsyncTask;
+import android.support.annotation.NonNull;
+import android.support.annotation.Nullable;
+
+import java.lang.ref.WeakReference;
+import java.util.HashMap;
+
+import fr.neamar.kiss.KissApplication;
+import fr.neamar.kiss.utils.UserHandle;
+
+public class MemoryCacheHelper {
+
+    private static final HashMap<AppIconHandle, Drawable> sAppIconCache = new HashMap<>();
```

Comment:
You may want to use the thread-safe `ConcurrentHashMap`.

Reply:
Nope! I want to sync on my terms and it's private for a reason.